### DISCUSSION TITLE
ActiveRecord::Base.connection.pool.lock_thread set to true but never restored causing problems in feature testing

### DISCUSSION THREAD

**Discussion by joeywang** [test-prof/test-prof#218] [General]
```
ActiveRecord::Base.connection.pool.lock_thread = true
```
This locked the connection.pool when you are trying to do feature testing and start puma with Capybara the connections number will stay as 1 instead of 2. This lead to error like the below:
```
Cannot expire connection, it is owned by a different thread: #<Thread:0x0000563303f39ee0@puma threadpool 002>
```

**Comment by palkan** [comment]
Could you please provide more details (or reproduction example): which TestProf tools do you use? When this exception occurs (backtrace)?

**Comment by joeywang** [comment]
I'm really sorry, I forget about this.
When we do feature testing start with more than one threads from test puma server if we lock the thread there will be a chance one thread is trying to grab the connection but not release by the current one.
Our fix to this is to reset the ActiveRecord::Base.connection.pool.lock_thread  to be false after each test.

```
Failure/Error:
       raise ActiveRecordError, "Cannot expire connection, " \
         "it is owned by a different thread: #{@owner}. " \
         "Current thread: #{Thread.current}."
     
     ActiveRecord::ActiveRecordError:
       Cannot expire connection, it is owned by a different thread: #<Thread:0x00000000149332d0@puma threadpool 001@/home/travis/build/reallyenglish-global/rex/vendor/bundle/ruby/2.6.0/gems/puma-5.3.2/lib/puma/thread_pool.rb:103 sleep>. Current thread: #<Thread:0x00000000022cb2d8 run>.
```

**Reply by palkan** [reply]
Please, see this discussion: #232

**Comment by kennethkalmer** [comment]
The fix in #236 shows up in the Changelog as part of v1.0.8, but that version isn't tagged in GitHub and doesn't exist on Rubygems either. 

Any chance we can get it published, or the Changelog updated to have it show up as unreleased? 🙏

**Reply by palkan** [reply]
Oh, sorry for that, looks like publishing failed. Fixed, 1.0.8 is now available.
